File: src/gui/src-tauri/src/commands/logging.rs
use log::{debug, info, warn, error};

/// Log a debug message
/// # Arguments
/// * `msg` - Message to log
#[tauri::command]
pub fn debug(msg: String) {
  debug!("{}", msg)
}

/// Log an info message
/// # Arguments
/// * `msg` - Message to log
#[tauri::command]
pub fn info(msg: String) {
  info!("{}", msg)
}

/// Log a warning message
/// # Arguments
/// * `msg` - Message to log
#[tauri::command]
pub fn warn(msg: String) {
  warn!("{}", msg)
}

/// Log an error message
/// # Arguments
/// * `msg` - Message to log
#[tauri::command]
pub fn error(msg: String) {
  error!("{}", msg)
}

File: src/gui/src/services/backend/logging.ts
import {invoke} from '@tauri-apps/api/tauri'

export async function debug(msg: string) {
  await invoke('debug', {msg})
}

export async function info(msg: string) {
  await invoke('info', {msg})
}

export async function warn(msg: string) {
  await invoke('warn', {msg})
}

export async function error(msg: string) {
  await invoke('error', {msg})
}
